packages/loader/container-loader/src/container.ts

```diff
@@ -1805,7 +1801,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private async fetchSnapshotTree(specifiedVersion: string | undefined):
         Promise<{snapshot?: ISnapshotTree; versionId?: string}>
     {
-        const version = await this.getVersion(specifiedVersion ?? this.id);
+        const version = await this.getVersion(specifiedVersion ?? this._resolvedUrl?.id ?? "");
```

Comment:
Why ""? It will not work, so maybe just fail more explicitly here?
That said, I think (worth double checking) that null means same thing as using this._resolvedUrl?.id, i.e. use latest version for this doc. If so, I'd use that everywhere in our code and remove (over time) usage of docId as some special value for this API.

Reply:
i just copied `this._resolvedUrl?.id ?? ""` from the original definition of container.id
right now this._resolvedUrl?.id is `string | undefined`, not the same with `string | null`

Reply:
But I believe getVersion("") will fail. So that does not look like correct flow.
if null means latest, then I'd say we should use getVersion(specifiedVersion ?? null) instead

Reply:
it seems specifiedVersion ?? null has failed some snapshot tests, with error "Attempted to access context before it was defined"
specifiedVersion ?? this._resolvedUrl?.id ?? null will work

Reply:
Can you please take a closer look why it fails and if it's a simple fix?
For example, I see that FluidFetchReader.getVersions() does not handle null correctly. Same for FileSnapshotWriterClassFactory(). I'd think that's the reason for failures (though I do not see how).

Maybe we want to have a separate PR (ahead of this one) that goes over all implementations of IDocumentStorageService.getVersions and validates that they correctly implement interface. I'm actually surprised that compiler does not complain about it, as it should not allow such casting / scope change. Any chance you can glance if we disable any warnings and should re-enable them?